tui/menu_button.go

```diff
@@ -0,0 +1,235 @@
+// Copyright © 2018 Intel Corporation
+//
+// SPDX-License-Identifier: GPL-3.0-only
+
+package tui
+
+import (
+	"fmt"
+
+	"github.com/VladimirMarkelov/clui"
+	xs "github.com/huandu/xstrings"
+	term "github.com/nsf/termbox-go"
+	"sync/atomic"
+	"time"
+)
+
+// MenuButton is the implementation of a clui button with simpler
+// visual elements (i.e no shadows).
+type MenuButton struct {
+	clui.BaseControl
+	settingsValue string
+	pressed       int32
+	onClick       func(clui.Event)
+	status        int
+	sWdith        int // splitter width
+}
+
+const (
+	textPadding = 3
+
+	// MenuButtonStatusDefault means: not the auto detect,
+	// not the user defined, not even failed
+	MenuButtonStatusDefault = iota
+
+	// MenuButtonStatusAutoDetect means: the current value was autodetected
+	MenuButtonStatusAutoDetect
+
+	// MenuButtonStatusUserDefined means: the user actively changed the value
+	MenuButtonStatusUserDefined
+
+	//MenuButtonStatusFailure means: the label displayed is an error message
+	MenuButtonStatusFailure
+)
+
+var (
+	// by now we're using the same char for everything
+	statusSymbol = map[int]rune{
```

Comment:
because I haven't decided yet how to differentiate , and for now I'm considering a more "textual" solution rather than a symbol and discrete alternative.

Reply:
I'll leave it that way, we adapt the code accordingly if we change the UI definitions.